database/Listing.js

```diff
@@ -0,0 +1,16 @@
+const mongoose = require("mongoose");
+const db = require("./index.js");
+const Image = require("./Image.js");
+mongoose.Promise = global.Promise;
+
+const listingSchema = new mongoose.Schema({
+  documentId: { type: Number, default: 0 },
```

Comment:
In order to avoid duplicate records you can make one of the properties unique ( { unique: true} ). I would make documentId unique.